agent/ebs/watcher.go

```diff
@@ -244,6 +248,10 @@ func (w *EBSWatcher) notifyAttachedEBS(volumeId string) {
 		log.Errorf("Unable to find EBS volume with volume ID: %v within agent state.", volumeId)
 		return
 	}
+	if !ebs.IsAttached() {
```

Comment:
`notifyAttachedEBS` might be called for a volume that's not attached? That sounds odd to me.

Reply:
Currently we're calling both "stage all" and "notify attached" for all our found attachments.  These will only execute if the attachment is in the right state -- so we only stage unstaged, un-notified volumes and we only notify on staged volumes.  The rest are no-ops.